internal/async/goroutine.go

```diff
@@ -30,7 +30,7 @@ func IsMainGoroutine() bool {
 //
 // This will be removed later and should never be public
 func EnsureNotMain(fn func()) {
-	if build.DisableThreadChecks || !IsMainGoroutine() {
+	if (build.MigratedToFyneDo() || !build.HasHints) || !IsMainGoroutine() {
```

Comment:
You can remove the unnecessary parenthesis, assuming you did not mean to have an `&&` in there?